overlord/snapstate/handlers.go

```diff
@@ -1377,12 +1385,25 @@ func (m *SnapManager) doUnlinkCurrentSnap(t *state.Task, _ *tomb.Tomb) (err erro
 	// symlink to a new revision of the snapd snap, so only do the actual
 	// unlink if we're not working on the snapd snap
 	if oldInfo.Type() != snap.TypeSnapd {
+		var reason unlinkReason
+		if err := t.Get("unlink-reason", &reason); err != nil && !errors.Is(err, state.ErrNoState) {
+			return err
+		}
+		experimentalRefreshAppAwarenessUX, err := features.Flag(tr, features.RefreshAppAwarenessUX)
+		if err != nil && !config.IsNoOption(err) {
+			return err
+		}
+		skipBinaries := false
+		if reason == unlinkReasonRefresh && refreshAppAwarenessEnabled && experimentalRefreshAppAwarenessUX {
+			skipBinaries = true
+		}
```

Comment:
I don't think we need the if here
```suggestion
		skipBinaries := reason == unlinkReasonRefresh && refreshAppAwarenessEnabled && experimentalRefreshAppAwarenessUX
```

Reply:
updated, thank you!